diff -ru ucspi-ipc-0.67.orig/src/Makefile ucspi-ipc-0.67/src/Makefile
--- ucspi-ipc-0.67.orig/src/Makefile	2013-10-07 15:42:01.923987612 +0200
+++ ucspi-ipc-0.67/src/Makefile	2013-10-07 15:43:44.569948650 +0200
@@ -197,8 +197,7 @@
 
 hasshsgr.h: choose compile tryshsgr.c hasshsgr.h1 hasshsgr.h2 chkshsgr \
 warn-shsgr
-	./chkshsgr || ( cat warn-shsgr; exit 1 )
-	./choose clr tryshsgr hasshsgr.h1 hasshsgr.h2 > hasshsgr.h
+	cat hasshsgr.h1 > hasshsgr.h
 
 haswaitp.h: choose compile load trywaitp.c haswaitp.h1 haswaitp.h2
 	./choose cl trywaitp haswaitp.h1 haswaitp.h2 >haswaitp.h
