diff -ru qmail-1.03.orig/Makefile qmail-1.03/Makefile
--- qmail-1.03.orig/Makefile	1998-06-15 12:53:16.000000000 +0200
+++ qmail-1.03/Makefile	2013-09-30 14:20:45.839183598 +0200
@@ -678,12 +678,7 @@
 
 hasshsgr.h: \
 chkshsgr warn-shsgr tryshsgr.c compile load
-	./chkshsgr || ( cat warn-shsgr; exit 1 )
-	( ( ./compile tryshsgr.c \
-	&& ./load tryshsgr && ./tryshsgr ) >/dev/null 2>&1 \
-	&& echo \#define HASSHORTSETGROUPS 1 || exit 0 ) > \
-	hasshsgr.h
-	rm -f tryshsgr.o tryshsgr
+	touch $@
 
 haswaitp.h: \
 trywaitp.c compile load
