diff -u mailfront-addons-src/INSTHIER.local mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/INSTHIER.local
--- mailfront-addons-src/INSTHIER.local	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/INSTHIER.local	2018-04-10 14:08:41.100502902 +0200
@@ -3,9 +3,6 @@
 c:::755::plugin-batv.so
 c:::755::plugin-dcc.so
 c:::755::plugin-greylist.so
-c:::755::plugin-spamassassin.so
-c:::755::plugin-sqlog.so
+c:::755::plugin-sauser.so
 c:::755::plugin-chkdns.so
 c:::755::plugin-authres.so
-c:::755::plugin-arlog.so
-c:::755::sqllib.so
diff -u mailfront-addons-src/Makefile.local mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/Makefile.local
--- mailfront-addons-src/Makefile.local	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/Makefile.local	2018-04-10 09:09:04.285305523 +0200
@@ -9,31 +9,32 @@
 
 all:  backend-qmailsump.so \
 	plugin-greylist.so plugin-dcc.so plugin-sauser.so plugin-batv.so \
-	plugin-sqlog.so plugin-arlog.so plugin-chkdns.so sqllib.so plugin-authres.so
+	plugin-chkdns.so plugin-authres.so
+#	plugin-sqlog.so plugin-arlog.so plugin-chkdns.so sqllib.so plugin-authres.so
 
 backend-qmailsump.so: makeso backend-qmailsump.c mailfront.h responses.h constants.h conf_qmail.c
-	./makeso backend-qmailsump.c  -lbg -lbg-sysdeps 
+	./makeso backend-qmailsump.c  -lbg
 
 plugin-batv.so: makeso plugin-batv.c mailfront.h responses.h constants.h
-	./makeso plugin-batv.c  -lbg -lbg-sysdeps -lcrypto
+	./makeso plugin-batv.c  -lbg -lcrypto
 
 plugin-dcc.so: makeso plugin-dcc.c mailfront.h responses.h constants.h
-	./makeso plugin-dcc.c  -lbg -lbg-sysdeps 
+	./makeso plugin-dcc.c  -lbg
 
 plugin-greylist.so: makeso plugin-greylist.c mailfront.h responses.h constants.h
-	./makeso plugin-greylist.c  -lbg -lbg-sysdeps 
+	./makeso plugin-greylist.c  -lbg
 
 plugin-sauser.so: makeso plugin-sauser.c mailfront.h responses.h constants.h
-	./makeso plugin-sauser.c  -lbg -lbg-sysdeps 
+	./makeso plugin-sauser.c  -lbg
 
 plugin-chkdns.so: makeso plugin-chkdns.c mailfront.h responses.h constants.h
-	./makeso plugin-chkdns.c  -lbg -lbg-sysdeps 
+	./makeso plugin-chkdns.c  -lbg -lresolv
 
 plugin-sqlog.so: makeso plugin-sqlog.c sqllib.so mailfront.h responses.h constants.h
 	./makeso plugin-sqlog.c ${CONFMODULES}/sqllib.so -lbg -lbg-sysdeps
 
 plugin-authres.so: makeso plugin-authres.c mailfront.h responses.h constants.h
-	./makeso plugin-authres.c -lbg -lbg-sysdeps -lspf2 -lopendkim -lopendmarc
+	./makeso plugin-authres.c -lbg -lspf2 -lopendkim -lopendmarc
 
 plugin-arlog.so: makeso plugin-arlog.c sqllib.so mailfront.h responses.h constants.h
 	./makeso plugin-arlog.c ${CONFMODULES}/sqllib.so -lbg -lbg-sysdeps \
@@ -47,5 +48,19 @@
 	bg-installer -v <INSTHIER.local
 	bg-installer -c <INSTHIER.local
 
+conf_qmail.c: conf-qmail
+	head -n 1 conf-qmail | \
+	  sed -e 's/"/\\"/g' \
+	      -e 's/^/const char conf_qmail[] = "/' \
+	      -e 's/$$/";/' >conf_qmail.c
+
+makeso: conf-ccso conf-ld
+	( echo '#!/bin/sh'; \
+	  echo 'source=$$1; shift'; \
+	  echo 'base=`echo "$$source" | sed -e s:\\\\.c$$::`'; \
+	  echo exec `head -n 1 conf-ccso` -DSHARED -I. -L. '-o $${base}.so $$source $${1+"$$@"}'; \
+	) >makeso
+	chmod 755 makeso
+
 clean: TARGETS.local
 	rm -f `cat TARGETS.local`
diff -u mailfront-addons-src/backend-qmailsump.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/backend-qmailsump.c
--- mailfront-addons-src/backend-qmailsump.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/backend-qmailsump.c	2018-04-10 08:55:48.965938102 +0200
@@ -5,15 +5,15 @@
  * SUMPADDR: address of the sump
  * RBLSMTPD: failure message for sump faux failure
  */
-#include <sysdeps.h>
+#include <bglibs/sysdeps.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <unistd.h>
-#include <misc/misc.h>
-#include <msg/msg.h>
-#include <unix/sig.h>
+#include <bglibs/misc.h>
+#include <bglibs/msg.h>
+#include <bglibs/sig.h>
 #include "mailfront.h"
 #include "conf_qmail.c"
 
diff -u mailfront-addons-src/plugin-arlog.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-arlog.c
--- mailfront-addons-src/plugin-arlog.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-arlog.c	2018-04-10 09:03:13.240388027 +0200
@@ -48,9 +48,9 @@
 #include <string.h>
 #include "mailfront.h"
 #include "conf_qmail.c"
-#include <iobuf/ibuf.h>
-#include <iobuf/obuf.h>
-#include <msg/msg.h>
+#include <bglibs/ibuf.h>
+#include <bglibs/obuf.h>
+#include <bglibs/msg.h>
 /* HACK HACK */
 #undef CLOCK_REALTIME
 #undef CLOCK_MONOTONIC
diff -u mailfront-addons-src/plugin-authres.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-authres.c
--- mailfront-addons-src/plugin-authres.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-authres.c	2018-04-10 14:03:05.018744074 +0200
@@ -15,12 +15,11 @@
 #include <string.h>
 #include "mailfront.h"
 #include "conf_qmail.c"
-#include <iobuf/ibuf.h>
-#include <iobuf/obuf.h>
-#include <msg/msg.h>
-#include <str/str.h>
-#include <dict/dict.h>
-#include <dict/load.h>
+#include <bglibs/ibuf.h>
+#include <bglibs/obuf.h>
+#include <bglibs/msg.h>
+#include <bglibs/str.h>
+#include <bglibs/dict.h>
 
 #include <opendkim/dkim.h>
 
@@ -29,8 +28,8 @@
 # include <netinet/in.h>   /* inet_ functions / structs */
 # include <arpa/inet.h> /* in_addr struct */
 
-#include <spf2/spf.h>
 #include <opendmarc/dmarc.h>
+#include <spf2/spf.h>
 
 static str arstr = {0,0,0};		/* authentication results header */
 
diff -u mailfront-addons-src/plugin-batv.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-batv.c
--- mailfront-addons-src/plugin-batv.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-batv.c	2018-04-10 09:20:36.192066300 +0200
@@ -14,12 +14,11 @@
 #include <ctype.h>
 #include "mailfront.h"
 #include "conf_qmail.c"
-#include <iobuf/ibuf.h>
-#include <dict/dict.h>
-#include <dict/load.h>
-#include <msg/msg.h>
+#include <bglibs/ibuf.h>
+#include <bglibs/dict.h>
+#include <bglibs/msg.h>
 #include <openssl/md5.h>
-#include <sys/time.h>
+#include <bglibs/systime.h>
 #include <string.h>
 
 #define BATVLEN 3		/* number of bytes */
diff -u mailfront-addons-src/plugin-chkdns.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-chkdns.c
--- mailfront-addons-src/plugin-chkdns.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-chkdns.c	2018-04-10 09:21:01.448203351 +0200
@@ -18,7 +18,7 @@
 #include <arpa/nameser.h>
 #include <resolv.h>
 
-#include <msg/msg.h>
+#include <bglibs/msg.h>
 
 static RESPONSE(badfrom,553,"5.1.8 Invalid sender domain.");
 static response resp_baddbl = { 553, "???" };
diff -u mailfront-addons-src/plugin-dcc.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-dcc.c
--- mailfront-addons-src/plugin-dcc.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-dcc.c	2018-04-10 09:09:47.985543592 +0200
@@ -6,10 +6,10 @@
 #include <unistd.h>
 #include <string.h>
 #include "mailfront.h"
-#include <net/socket.h>
-#include <iobuf/ibuf.h>
-#include <iobuf/obuf.h>
-#include <msg/msg.h>
+#include <bglibs/socket.h>
+#include <bglibs/ibuf.h>
+#include <bglibs/obuf.h>
+#include <bglibs/msg.h>
 
 static str dccsender;
 static str dccrecips;
diff -u mailfront-addons-src/plugin-greylist.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-greylist.c
--- mailfront-addons-src/plugin-greylist.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-greylist.c	2018-04-10 09:01:08.293702468 +0200
@@ -10,7 +10,7 @@
 #include <unistd.h>
 
 #include "mailfront.h"
-#include <net/socket.h>
+#include <bglibs/socket.h>
 
 static str greymsg;
 static int hasgreyrcpt = 0;
diff -u mailfront-addons-src/plugin-sauser.c mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-sauser.c
--- mailfront-addons-src/plugin-sauser.c	2013-05-31 17:14:21.000000000 +0200
+++ mailfront-addons-build/mailfront-addons-2467b672b84aaf13b61905885c7ffa185cd86ae8-src/plugin-sauser.c	2018-04-10 09:13:16.592678750 +0200
@@ -9,13 +9,13 @@
 #include <unistd.h>
 #include <string.h>
 #include "mailfront.h"
-#include <net/socket.h>
-#include <iobuf/ibuf.h>
-#include <iobuf/obuf.h>
-#include <msg/msg.h>
+#include <bglibs/socket.h>
+#include <bglibs/ibuf.h>
+#include <bglibs/obuf.h>
+#include <bglibs/msg.h>
 #include "conf_qmail.c"
-#include <dict/dict.h>
-#include <dict/load.h>
+#include <bglibs/dict.h>
+//#include <bglibs/load.h>
 
 static dict sanf;
 static str sasender;
