diff -ru mailfront-addons-orig/plugin-authres.c mailfront-addons-build/plugin-authres.c
--- mailfront-addons-orig/plugin-authres.c	2018-04-11 08:25:38.104345911 +0200
+++ mailfront-addons-build/plugin-authres.c	2018-04-12 20:24:50.879188589 +0200
@@ -6,6 +6,7 @@
  * Check SPF, too
  * env DMARCREJECT=y means actually do reject
  * file control/nodmarcpolicy lists domains not to reject
+ * Check is skipped if RELAYCLIENT is set or user is authenticated.
  *
 */
 
@@ -52,6 +53,9 @@
 	ip = getprotoenv("REMOTEIP");
 	if(!ip) return 0;		/* can't tell IP, no SPF */
 
+	if (session_getenv("RELAYCLIENT") != 0 || session_getnum("authenticated", 0))
+		return 0;
+
 	spf_server = SPF_server_new(SPF_DNS_CACHE, 0);
 	if(!spf_server) return 0;
 	spf_request = SPF_request_new(spf_server);
@@ -71,7 +75,12 @@
 	str_cat2s(&arstr, "; spf=", SPF_strresult(SPF_response_result(spf_response)));
 
 	str_cat4s(&arstr, " spf.mailfrom=", sender->s, " spf.helo=", helo);
-	
+
+	session_setenv("PLUGIN_AUTHRES_SPF_RESULT",
+		SPF_strresult(SPF_response_result(spf_response)), 1);
+	session_setenv("PLUGIN_AUTHRES_SPF_MAILFROM", sender->s, 1);
+	session_setenv("PLUGIN_AUTHRES_SPF_HELO", helo, 1);
+
 	/* for DMARC later */
 	if(!spf_domain.s) str_init(&spf_domain);
 	str_copys(&spf_domain, sender->s);
@@ -123,6 +132,9 @@
 	str fromdom;             /* from domain name */
 	const char *qh;
 
+	if (session_getenv("RELAYCLIENT") != 0 || session_getnum("authenticated", 0))
+		return 0;
+
 	if(!authservid) authservid = getprotoenv("LOCALHOST");
 	if(!authservid) authservid = "localhost";
 
@@ -275,6 +287,9 @@
 				  " policy=",dmpol);
 
 			msg6("dmarc: ",dmres," for ", fromdom.s, " policy=",dmpol);
+			session_setenv("PLUGIN_AUTHRES_DMARC_RESULT", dmres, 1);
+			session_setenv("PLUGIN_AUTHRES_DMARC_HEADER_FROM", fromdom.s, 1);
+			session_setenv("PLUGIN_AUTHRES_DMARC_POLICY", dmpol, 1);
 		}
 
 		/* do this only so many percent */
