diff -ru mailfront-orig/plugin-cvm-authenticate.c mailfront-build/plugin-cvm-authenticate.c
--- mailfront-orig/plugin-cvm-authenticate.c	2015-02-12 15:10:37.000000000 +0100
+++ mailfront-build/plugin-cvm-authenticate.c	2018-04-11 12:28:48.546472322 +0200
@@ -31,16 +31,24 @@
   }
   else {
     session_setnum("authenticated", 1);
+    session_setenv("PLUGIN_CVM_AUTH_AUTHENTICATED", "1", 0);
     session_delstr("helo_domain");
     session_setstr("auth_user", cvm_fact_username);
+    session_setenv("PLUGIN_CVM_AUTH_USER", cvm_fact_username, 0);
     session_setnum("auth_uid", cvm_fact_userid);
     session_setnum("auth_gid", cvm_fact_groupid);
-    if (cvm_fact_realname != 0)
+    if (cvm_fact_realname != 0) {
       session_setstr("auth_realname", cvm_fact_realname);
-    if (cvm_fact_domain != 0)
+      session_setenv("PLUGIN_CVM_AUTH_REALNAME", cvm_fact_realname, 0);
+    }
+    if (cvm_fact_domain != 0) {
       session_setstr("auth_domain", cvm_fact_domain);
-    if (cvm_fact_mailbox != 0)
+      session_setenv("PLUGIN_CVM_AUTH_DOMAIN", cvm_fact_domain, 0);
+    }
+    if (cvm_fact_mailbox != 0) {
       session_setstr("auth_mailbox", cvm_fact_mailbox);
+      session_setenv("PLUGIN_CVM_AUTH_MAILBOX", cvm_fact_mailbox, 0);
+    }
     respond(&resp_authenticated);
   }
   return 1;
